library/src/main/scala/scala/scalajs/LinkingInfo.scala

```diff
@@ -108,4 +108,35 @@ object LinkingInfo {
   def assumingES6: Boolean =
     linkingInfo.assumingES6
 
+  /** The value of the global JavaScript `this`.
+   *
+   *  This returns the value that would be obtained by writing `this` at the
+   *  top-level of the JavaScript file generated by Scala.js. In most
+   *  JavaScript environments, this is equivalent to the *global object*, but
+   *  it is not necessarily the case. For example, on Node.js, `this` is the
+   *  object representing the `exports` of the current module.
+   *
+   *  Using this value should be rare, and mostly limited to writing code
+   *  detecting what the global object is. For example, a typical detection
+   *  code looks like:
+   *  {{{
+   *  val globalObject = {
+   *    import js.Dynamic.{global => g}
+   *    if (js.typeOf(g.global) != "undefined" && (g.global.Object eq g.Object)) {
+   *      // Node.js environment detected
+   *      g.global
+   *    } else {
+   *      // In all other well-known environment, we can use the global `this`
+   *      scala.scalajs.LinkingInfo.globalThis
+   *    }
+   *  }
+   *  }}}
+   *  Note that the above code is not comprehensive, as there can be JavaScript
+   *  environments where the global object cannot be fetched neither through
+   *  `global` nor `this`. If your code needs to run in such an environment, it
+   *  is up to you to use an appropriate detection procedure.
+   */
+  @inline
+  def globalThis: js.Dynamic = linkingInfo.globalThis
+
```

Comment:
I have thought about that for the past two days, and also brainstormed a bit with @nicolasstucki this morning.

My proposal would be `js.special._`. It reflects the fact that the tools in there are quite special and should not be used on a regular basis.

It also does not have the connotation of `js.completeness._` that's it's there "for theoretical completeness" only (they are sometimes required in practice). `js.support._` was not bad, but it implies that the tools there are used to build other tools only, which is not the case of `debugger` or `newTarget`. I had discarded `BuildingBlocks` for the same reason (not to mention it's ugly).